python/test/test_api.py

```diff
@@ -311,14 +338,51 @@ def test_invalid_repo(self):
             tritonserver.Server(model_repository="foo").start()
 
     def test_ready(self):
-        server = tritonserver.Server(server_options).start()
+        server = tritonserver.Server(self._server_options).start()
+        self.assertTrue(server.ready())
+
+    def test_stop(self):
+        server = tritonserver.Server(self._server_options).start(wait_until_ready=True)
+
         self.assertTrue(server.ready())
 
+        server.load(
+            "test",
+            {
+                "config": json.dumps(
+                    {
+                        "backend": "python",
+                        "parameters": {"decoupled": {"string_value": "False"}},
+                    }
+                )
+            },
+        )
+
+        fp16_input = numpy.random.rand(1, 100).astype(dtype=numpy.float16)
+
+        for response in server.model("test").infer(
+            inputs={"fp16_input": fp16_input},
+            output_memory_type="cpu",
+            raise_on_error=True,
+        ):
+            fp16_output = numpy.from_dlpack(response.outputs["fp16_output"])
+            numpy.testing.assert_array_equal(fp16_input, fp16_output)
+        try:
+            server.stop()
+        except tritonserver.InternalError as e:
+            pytest.xfail(f"Server failed to stop! {e}")
```

Comment:
I think it would be easier to find/analyze the failures for debugging if we let the test fail on gitlab side of things. I think we can let the test fail as a known failure for release purposes.

Reply:
updated to use xfail only for versions <= "2.42.0" to indicate the test is expected to fail / is a known issue.